src/main/java/gift/controller/KakaoOauthController.java

```diff
@@ -0,0 +1,72 @@
+package gift.controller;
+import gift.common.dto.response.TokenResponseDto;
+import gift.external.kakao.response.GetMemberIdResponseDto;
+import gift.external.kakao.response.GetTokenResponseDto;
+import gift.external.kakao.KakaoApiClient;
+import gift.service.MemberService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+
+@RestController
+@RequestMapping("/api/oauth/kakao")
+public class KakaoOauthController {
+
+    @Value("${kakao.client-id}")
+    private String clientId;
+
+    @Value("${kakao.redirect-register}")
+    private String redirectRegister;
+
+    @Value("${kakao.redirect-login}")
+    private String redirectLogin;
+
+    private final KakaoApiClient kakaoClient;
+
+    private final MemberService memberService;
+
+    public KakaoOauthController(KakaoApiClient kakaoClient, MemberService memberService) {
+        this.kakaoClient = kakaoClient;
+        this.memberService = memberService;
+    }
+
+    @GetMapping("/register/code")
+    public ResponseEntity<Void> registerCode() {
+        return ResponseEntity.status(HttpStatus.TEMPORARY_REDIRECT)
+                .header("Location", getKakaoAuthUrl(redirectRegister))
```

Comment:
```suggestion
                .header(HttpHeaders.LOCATION, getKakaoAuthUrl(redirectRegister))
```
요렇게하면 실수할 일이 줄어들 것 같네요😊